src/DataObjects/WorkflowInstance.php

```diff
@@ -585,7 +585,8 @@ protected function userHasAccess($member)
         // This method primarily "protects" access to a WorkflowInstance, but assumes access only to be granted to
         // users assigned-to that WorkflowInstance. However; lowly authors (users entering items into a workflow) are
         // not assigned - but we still wish them to see their submitted content.
-        $inWorkflowGroupOrUserTables = ($member->inGroups($this->Groups()) || $this->Users()->find('ID', $member->ID));
+        $inWorkflowGroupOrUserTables = $this->Users()->filter('ID', $member->ID)->exists()
+            || $member->inGroups($this->Groups());
```

Comment:
I've also changed the order of these - it doesn't change the outcome, but the users query will be faster than the `inGroups()` check, so doing that first means sometimes we can skip the slower check.